frame/contracts/src/wasm/runtime.rs

```diff
@@ -508,24 +589,24 @@ define_env!(Env, <E: Ext>,
 	//   Should be decodable as a `T::Balance`. Traps otherwise.
 	// - value_len: length of the value buffer.
 	//
-	// # Traps
+	// # Errors
```

Comment:
Above it states that it traps if the value is not decodable. Is it the case still?

Reply:
Yes. Decoding and out of bound accesses always trap. You can look that up in `err_into_return_code` that only a selection of errors are non fatal and return as `Ok(code)`. The rest is a `Err` and will trap.